docs/core/whats-new/index.md

```diff
@@ -1,159 +1,179 @@
 
-### Support for .NET Framework libraries
+## API changes 
 
-.NET Core code can reference existing .NET Framework libraries, including existing NuGet packages. Note that the libraries must use APIs that are found in .NET Standard.
+### `Span<T>` and `Memory<T>`
 
-## Visual Studio integration
+[!INCLUDE[.NET Core 2.1](~/includes/net-core-21.md)] includes some new types that make working with arrays and other types of memory much more efficient. The new types include:
 
-Visual Studio 2017 version 15.3 and in some cases Visual Studio for Mac offer a number of significant enhancements for .NET Core developers.
+- <xref:System.Span%601?displayProperty=nameWithType> and <xref:System.ReadOnlySpan%601?displayProperty=nameWithType>.
 
-### Retargeting .NET Core apps and .NET Standard libraries
+- <xref:System.Memory%601?displayProperty=nameWithType> and <xref:System.ReadOnlyMemory%601>.
 
-If the .NET Core 2.0 SDK is installed, you can retarget .NET Core 1.x projects to .NET Core 2.0 and .NET Standard 1.x libraries to .NET Standard 2.0.
+Without these types, when passing such items as a portion of an array or a section of a memory buffer, you have to make a copy of some portion of the data before passing it to a method. These types provide a virtual view of that data which eliminates the need for the additional memory allocation and copy operations.
 
-To retarget your project in Visual Studio, you open the **Application** tab of the project's properties dialog and change the **Target framework** value to **.NET Core 2.0** or **.NET Standard 2.0**. You can also change it by right-clicking on the project and selecting the **Edit \*.csproj file** option. For more information, see the [Tooling](#tooling) section earlier in this topic.
+The following example uses a <xref:System.Span%601> instance to provide a virtual view of ten elements of an array.
 
-### Live Unit Testing support for .NET Core
+```csharp
+using System;
 
-Whenever you modify your code, Live Unit Testing automatically runs any affected unit tests in the background and displays the results and code coverage live in the Visual Studio environment. .NET Core 2.0 now supports Live Unit Testing. Previously, Live Unit Testing was available only for .NET Framework applications.
+class Program
+{
+    static void Main(string[] args)
+    {
+        int[] numbers = new int[100];
+        for (int ctr = 0; ctr < 100; ctr++)
+        {
+           numbers[ctr] = (ctr + 1) * 2;  
+        } 
 
-For more information, see [Live Unit Testing with Visual Studio 2017](/visualstudio/test/live-unit-testing) and the [Live Unit Testing FAQ](/visualstudio/test/live-unit-testing-faq).
+        var part = new Span<int>(numbers, 10, 10);
+        foreach (var value in part)
+           Console.Write($"{value}  ");
+    }
+}
+// The example displays the following output:
+//      2  24  26  28  30  32  34  36  38  4
+```
```

Comment:
1. The code actually prints `22  24  26  28  30  32  34  36  38  40`.
2. Since `args` is not used, it can be removed.
3. I think `ctr` is not a clear name. Is there a reason why the conventional `i` isn't used?
4. Instead of `(ctr + 1) * 2`, why not just use `ctr * 2`? It changes the output, but makes the code simpler and more focused on what's important.
5. The code has inconsistent indentation: most of the time, it uses 4 spaces, but sometimes 3.
6. The meaning of the numbers in the `Span` constructor might be unclear. Maybe it would be better to use named arguments?

Putting it all together, the code would become:

```c#
using System;

class Program
{
    static void Main()
    {
        int[] numbers = new int[100];
        for (int i = 0; i < 100; i++)
        {
            numbers[i] = i * 2;
        }

        var part = new Span<int>(numbers, start: 10, length: 10);
        foreach (var value in part)
            Console.Write($"{value}  ");
    }
}
// The example displays the following output:
//     20  22  24  26  28  30  32  34  36  38
```